File: Swift-C++-Testsuite/test/concurrency/Swift/ActorExtension.swift
@main
struct MyMain {

    static func main() async {
        let logger = TemperatureLogger(label: "Outdoors", measurement: 25)
        let maxTemp = await logger.max
        logger.update(with: 25)
        let newMaxTemp = await logger.max
    }

}

actor TemperatureLogger {
    let label: String
    var measurements: [Int]
    private(set) var max: Int

    init(label: String, measurement: Int) {
        self.label = label
        self.measurements = [measurement]
        self.max = measurement
    }
}

extension TemperatureLogger {
    func update(with measurement: Int) {
        measurements.append(measurement)
        if measurement > max {
            max = measurement
        }
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-1/ActorExtension.cpp

#include <iostream>
#include <vector>
#include <algorithm>

class TemperatureLogger {
public:
    std::string label;
    std::vector<int> measurements;
    int max;

    TemperatureLogger(std::string label, int measurement) : label(label), measurements(std::vector<int>{measurement}), max(measurement) {};

    void update(int measurement){
        measurements.push_back(measurement);
        max = std::max(max, measurement);
    }
};

int main()
{
    TemperatureLogger logger("Outdoors", 25);
    int maxTemp = logger.max;
    logger.update(25);
    int newMaxTemp = logger.max;    

    return 0;
}